commonui/src/main/java/com/hyperwallet/android/ui/common/view/OnItemClickListener.java

```diff
@@ -0,0 +1,25 @@
+/*
```

Comment:
I don't think we should use this approach. At first, it seems like a good idea but when I look deeper at it, I have the following thoughts:
* The class implementing this interface is the upper class, the adapter, which is responsible for inflating the view holders and managing the data that is sent to the recycler view, AND handling the clicks on each view holder, while in fact, it should be the view holder itself responsible for handling its events. I understand that the view holder is invoking/delegating the event handling to the upper class which I think isn't the ideal approach. In my opinion, the view holder should handle the click and not delegate it to the upper layer. 
* Since the onBindViewHolder already gives us the position and an easy way to get the destination (transfer method object), it makes it more appealing to use a listener there and let each view holder being an independent "object" since each one of them will have its own listener as opposed to all view holder sharing the same listener (the adapter)
We can discuss this offline if needed

Reply:
yeah thats part of Epam's code simplified it :)